### DISCUSSION TITLE
Utilities with permissions

### DISCUSSION THREAD

**Discussion by HazemMohamed252** [statamic/cms#5697] [Q&A] [answered]
I just created new utility call notification
A single permission will be registered automatically using the handle.
eg. access notificatons utility

Where to add this line of permission please ?

**Comment by duncanmcclean** [comment]
Unlike nav items, utilities will handle permissions for you.

**Reply by HazemMohamed252** [reply]
No because as i did it and mark on it to be not shown from specific users ok it disappeared from the sidebar but when i access the URL it direct in right way and the user can access the page

**Comment by jasonvarga** [comment, marked as answer]
This looks like a bug, good find. Can you open an issue? The permission should apply to the page itself, too. (Assuming you are just using a view like the `Utility::make('french-fries')->view('fries-utility')->register();` example in the docs) If you use your own controller you'd have to write the authorize line yourself.

**Reply by HazemMohamed252** [reply]
Thanks
